docs/src/main/asciidoc/spring-cloud-config.adoc

```diff
@@ -318,6 +318,17 @@ system properties (`-Dhttps.proxyHost` and `-Dhttps.proxyPort`).
 
 TIP: If you do not know where your `~/.git` directory is, use `git config --global` to manipulate the settings (for example, `git config --global http.sslVerify false`).
 
+When working with SSH keys, the expected ssh private-key should begin with ```-----BEGIN RSA PRIVATE KEY-----```. If the SSH key starts with ```-----BEGIN OPENSSH PRIVATE KEY-----``` it will not work. The resulting error for an unreadable private key looks like:
+```
+- Error in object 'spring.cloud.config.server.git': codes [PrivateKeyIsValid.spring.cloud.config.server.git,PrivateKeyIsValid]; arguments [org.springframework.context.support.DefaultMessageSourceResolvable: codes [spring.cloud.config.server.git.,]; arguments []; default message []]; default message [Property 'spring.cloud.config.server.git.privateKey' is not a valid private key]
+```
+
+By default ssh-keygen generates an SSH key in a format that cannot be read by JGit. To correct for this scenario, generate a SSH key with the PEM format.
```

Comment:
I agree that we should mention this, but let's start with (ie move this above the previous two paragraphs) something like, "JGit requires RSA keys in PEM format. Below is an example ssh-keygen (from openssh) command:"